packages/storage/src/indexed-db/index.ts

```diff
@@ -802,4 +795,44 @@ export class IndexedDb implements IndexedDbInterface {
       output: Value.fromJson(result.output),
     };
   }
+
+  async fetchStakingTokenId(): Promise<AssetId> {
+    const registryClient = new ChainRegistryClient();
+    const registry = registryClient.get(this.chainId);
+    const stakingTokenMetadata = registry.getMetadata(registry.stakingAssetId);
+    const stakingToken = getAssetId(stakingTokenMetadata);
+
+    return stakingToken;
+  }
+
+  async hasStakingAssetBalance(assetId: AssetId): Promise<boolean> {
+    const spendableUMNotes = await this.db.getAllFromIndex(
+      'SPENDABLE_NOTES',
+      'assetId',
+      uint8ArrayToBase64(assetId.inner),
+    );
+
+    // Iterate over the spendable UM notes, and acrue balance for unspent notesz
+    let stakingTokenBalance = new Amount();
+    for (const note of spendableUMNotes) {
+      let umNote = note as unknown as SpendableNoteRecord;
+      if (umNote.heightSpent == undefined) {
```

Comment:
```suggestion
      if (umNote.heightSpent === undefined) {
```

However, if it's a class (as per my suggestion above of using `.fromJson()`), it will never be `undefined`, since Protobuf classes default to `0n` for undefined bigint values. So, hm, we'd need to think of another way to check this.

Reply:
you're right, then can't you just do `umNote.heightSpent === 0n` to filter for unspent notes?

Reply:
This is the thing that always confuses me a little bit about Protobuf's default values. `0n` is generally an actually meaningful value. It _happens_

Reply:
Whoops, bumped "Comment." 

... It _happens_ not to be meaningful in this specific use case, because it's impossible to have spent any notes at block height `0n` since that's the genesis block. So you could safely use that check, yes. But that's... weird and janky? I kind of wish Protobufs just allowed u64 values to be undefined, rather than defaulting to `0n`.

But tl;dr: yes, that check should work.